### ISSUE TITLE
Compile Error: Enum case value must be compile-time evaluatable -- Error v3

### ISSUE THREAD

**Issue by Wordlesschunk** [briannesbitt/Carbon#2934]
Hello,

I encountered an issue with the following code:
```php
echo Carbon::now()->startOfWeek();
```

Carbon version: 3.0.0

PHP version: 8.1.19 | Symfony 6.4.3

I expected to get:

```
 2024-01-29 00:00:00.0 UTC (+00:00)
```

But I actually get:
<img width="1045" alt="Screenshot 2024-02-03 at 15 00 00" src="https://github.com/briannesbitt/Carbon/assets/22942020/da7370b8-ff9c-4d40-8506-02ab1bc72cb8">
```
Compile Error: Enum case value must be compile-time evaluatable
```
PS:
Downgrading my version of Carbon to: 2.72.1 seems to resolve this issue
Thanks!

**Comment by kylekatarnls** [comment]
Hello, I tried to run your code with both 3.0.0 and 3.0.1, PHP 8.1 and 8.3. I can't happen to trigger this.

I suspect you just need to rebuild your Symfony cache.

If you would like I investigate further, please give a complete stack trace and a project example to reproduce the issue.

Thanks.

**Comment by kylekatarnls** [comment]
I checked where in PHP this error is thrown and I was able to get this error using this code instead:
```php
class_exists(\Carbon\WeekDay::class);
```

It only happen with PHP 8.1. As 8.2 fixes the issue and 8.1 is no longer in [active support](https://www.php.net/supported-versions.php), there is no chances that it got fixed in a PHP patch on 8.1.x. And little chances that we can work around it on Carbon side without degrading features.

So the best move when hitting this is to upgrade to PHP 8.2.
